File: archive-application/src/main/java/site/archive/service/banner/BannerService.java
package site.archive.service.banner;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import site.archive.domain.banner.Banner;
import site.archive.domain.banner.BannerRepository;
import site.archive.domain.banner.BannerType;
import site.archive.dto.v2.BannerListResponseDto;
import site.archive.dto.v2.BannerResponseDto;

@Service
@Transactional(readOnly = true)
@RequiredArgsConstructor
public class BannerService {

    private final BannerRepository bannerRepository;

    public BannerListResponseDto getAllBanner() {
        var bannerResponse = bannerRepository.findAllByOrderByCreatedAtDesc().stream()
                                             .map(banner -> new BannerResponseDto(banner.getType().toString(),
                                                                                  banner.getSummaryImage(),
                                                                                  banner.getMainContent()))
                                             .toList();
        return new BannerListResponseDto(bannerResponse);
    }

    @Transactional
    public BannerResponseDto createBanner(String summaryImageUri, String mainContent, BannerType type) {
        var banner = bannerRepository.save(new Banner(summaryImageUri, mainContent, type));
        return BannerResponseDto.from(banner);
    }

}

File: archive-application/src/main/kotlin/site/archive/service/banner/BannerService.kt
package site.archive.service.banner

import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import site.archive.domain.banner.Banner
import site.archive.domain.banner.BannerRepository
import site.archive.domain.banner.BannerType
import site.archive.dto.v2.BannerListResponseDto
import site.archive.dto.v2.BannerResponseDto

@Service
@Transactional(readOnly = true)
class BannerService(val bannerRepository: BannerRepository) {

    fun getAllBanner(): BannerListResponseDto {
        val bannerResponse = bannerRepository.findAllByOrderByCreatedAtDesc()
            .map { BannerResponseDto(it.type.toString(), it.summaryImage, it.mainContent) }
            .toList()
        return BannerListResponseDto(bannerResponse)
    }

    @Transactional
    fun createBanner(summaryImageUri: String, mainContent: String, type: BannerType): BannerResponseDto {
        val banner = bannerRepository.save(Banner(summaryImageUri, mainContent, type))
        return BannerResponseDto.from(banner)
    }

    @Transactional
    fun deleteBanner(bannerId: Long) {
        bannerRepository.deleteById(bannerId)
    }

}